examples/09_ui/main.rs

```diff
@@ -0,0 +1,226 @@
+//! Displays a shaded sphere to the user.
+
+extern crate amethyst;
+extern crate genmesh;
+
+use amethyst::assets::Loader;
+use amethyst::core::cgmath::Vector3;
+use amethyst::core::cgmath::prelude::InnerSpace;
+use amethyst::core::transform::Transform;
+use amethyst::ecs::World;
+use amethyst::prelude::*;
+use amethyst::renderer::{AmbientColor, Camera, DisplayConfig, DrawShaded, Light, Mesh, Pipeline,
+                         PngFormat, PointLight, PosNormTex, RenderBundle, RenderSystem, Rgba,
+                         ScreenDimensions, Stage};
+use amethyst::ui::{DrawUi, TtfFormat, UiBundle, UiImage, UiText, UiTransform};
+use amethyst::winit::{Event, KeyboardInput, VirtualKeyCode, WindowEvent};
+use genmesh::{MapToVertices, Triangulate, Vertices};
+use genmesh::generators::SphereUV;
+
+const SPHERE_COLOUR: [f32; 4] = [0.0, 0.0, 1.0, 1.0]; // blue
+const AMBIENT_LIGHT_COLOUR: Rgba = Rgba(0.01, 0.01, 0.01, 1.0); // near-black
+const POINT_LIGHT_COLOUR: Rgba = Rgba(1.0, 1.0, 1.0, 1.0); // white
+const BACKGROUND_COLOUR: [f32; 4] = [0.0, 0.0, 0.0, 0.0]; // black
+const LIGHT_POSITION: [f32; 3] = [2.0, 2.0, -2.0];
+const LIGHT_RADIUS: f32 = 5.0;
+const LIGHT_INTENSITY: f32 = 3.0;
+
+struct Example;
+
+impl State for Example {
+    fn on_start(&mut self, engine: &mut Engine) {
+        // Initialise the scene with an object, a light and a camera.
+        initialise_sphere(&mut engine.world);
+        initialise_lights(&mut engine.world);
+        initialise_camera(&mut engine.world);
+        let (logo, font) = {
+            let loader = engine.world.read_resource::<Loader>();
+
+            let logo = loader.load(
+                "texture/logo_transparent.png",
+                PngFormat,
+                Default::default(),
+                (),
+                &engine.world.read_resource(),
```

Comment:
Let's not call this method twice; please introduce a binding above.

Reply:
They're not the same thing, the inferred type is different between the calls.  I'll add the type.

Reply:
Huh, didn't notice that.